internal/api/authz.go

```diff
@@ -0,0 +1,60 @@
+package api
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/peer"
+)
+
+// roleMap defines the accessible methods for each role
+var roleMap = map[string][]string{
+	"/job.JobManager/Start":  {"admin"},
+	"/job.JobManager/Stop":   {"admin"},
+	"/job.JobManager/Status": {"admin", "user"},
+	"/job.JobManager/Output": {"admin", "user"},
+}
+
+// unaryInterceptor is a grpc inteceptor that authorizes access to the methods as listed in roleMap
+func unaryInterceptor(ctx context.Context, req any, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (any, error) {
+	// get the peer information so we can parse the client certificate out of it
+	peer, ok := peer.FromContext(ctx)
+	if !ok {
+		return nil, fmt.Errorf("error reading peer information from context")
```

Comment:
```suggestion
		return nil, errors.New("error reading peer information from context")
```

Prefer errors.New if you don't have any format verbs.